sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -333,6 +367,10 @@ class QueryExecution(
   }
 }
 
+object CommandExecutionMode extends Enumeration {
+  val COMMON, EAGERLY, SKIP = Value
```

Comment:
I'd make the name more explicit `SKIP`, `NON_ROOT`, `ALL`

Reply:
OK